### ISSUE TITLE
Optin error handling flowspace request

### ISSUE THREAD

**Issue by lbergesio** [fp7-ofelia/ocf#122]
[Wed Mar 27 11:02:04 2013] [error] [client 147.83.206.82]     return self.handle_uncaught_exception(request, resolver, sys.exc_info())
It may be when ranges (in this case MACs) are to big, or maybe when some limit is not specified.

LOG in optin:
[Wed Mar 27 11:02:04 2013] [error] [client 147.83.206.82]   File "/usr/lib/pymodules/python2.6/django/core/handlers/base.py", line 100, in get_response
[Wed Mar 27 11:02:04 2013] [error] [client 147.83.206.82]     response = callback(request, _callback_args, *_callback_kwargs)
[Wed Mar 27 11:02:04 2013] [error] [client 147.83.206.82]   File "/opt/ofelia/optin_manager/src/wsgi/openflow/optin_manager/../../../python/openflow/optin_manager/opts/views.py", line 180, in add_opt_in
[Wed Mar 27 11:02:04 2013] [error] [client 147.83.206.82]     selexp,request.user,assigned_priority,True)
[Wed Mar 27 11:02:04 2013] [error] [client 147.83.206.82]   File "/opt/ofelia/optin_manager/src/wsgi/openflow/optin_manager/../../../python/openflow/optin_manager/opts/helper.py", line 57, in opt_fs_into_exp
[Wed Mar 27 11:02:04 2013] [error] [client 147.83.206.82]     matches = range_to_match_struct(opt)
[Wed Mar 27 11:02:04 2013] [error] [client 147.83.206.82]   File "/opt/ofelia/optin_manager/src/wsgi/openflow/optin_manager/../../../python/openflow/optin_manager/flowspace/helper.py", line 247, in range_to_match_struct
[Wed Mar 27 11:02:04 2013] [error] [client 147.83.206.82]     for value in range(getattr(rangeFS,om_start), getattr(rangeFS, om_end)+1):
[Wed Mar 27 11:02:04 2013] [error] [client 147.83.206.82] OverflowError: range() result has too many items

**Comment by CarolinaFernandez** [comment]
This happens because Python's `range()` cannot iterate over a number of elements higher than `sys.maxsize` (which is the case for e.g. a big range of MACs, let's say 00:00:00:00:00:00 - 00:00:0F:FF:FF:FF).

Too many factors are to be taken into account to allow a range wide enough for every need, but also taking into account that Flowvisor heap space (Java heap space, currently set to 100M minimum, 1000M maximum) is limited and will be filled very soon. The used space also depends on the number of switches per island, because for every switch (datapath id) and MAC there will be a flowspace rule...

Summing up, it is allowed to have around 450 elements per range; to avoid Flowvisor complaining about its heap space after a (big) while and also to avoid waiting too much for a simple request.

If range is too big and/or Flowvisor does not have enough memory, an error will be raised to the Opt-in UI.
